model_compression_toolkit/data_generation/common/model_info_exctractors.py

```diff
@@ -0,0 +1,178 @@
+# limitations under the License.
+# ==============================================================================
+from typing import Type, Any, List
+
+
+class OrigBNStatsHolder(object):
+    """
+    Holds the original batch normalization (BN) statistics for a model.
+    """
+    def __init__(self,
+                 model: Any,
+                 bn_layer_types: Type[list],
+                 eps=1e-6):
+        """
+        Initializes the OrigBNStatsHolder.
+
+        Args:
+            model (Any): The framework model.
+            bn_layer_types (Type[list]): List of batch normalization layer types.
+            eps (float): Epsilon value for numerical stability.
+        """
+        self.eps = eps
+        self.bn_params = self.get_bn_params(model, bn_layer_types)
+
+    def get_bn_layer_names(self) -> List[str]:
+        """
+        Get the names of all batch normalization layers.
+
+        Returns:
+            List[str]: List of batch normalization layer names.
+        """
+        return list(self.bn_params.keys())
+
+    def get_mean(self, bn_layer_name: str) -> Any:
+        """
+        Get the mean of a batch normalization layer.
+
+        Args:
+            bn_layer_name (str): Name of the batch normalization layer.
+
+        Returns:
+            Any: Mean of the batch normalization layer.
+        """
+        return self.bn_params[bn_layer_name][0]
+
+    def get_var(self, bn_layer_name: str) -> Any:
+        """
+        Get the variance of a batch normalization layer.
+
+        Args:
+            bn_layer_name (str): Name of the batch normalization layer.
+
+        Returns:
+            Any: Variance of the batch normalization layer.
+        """
+        return self.bn_params[bn_layer_name][1]
+
+    def get_std(self, bn_layer_name: str) -> Any:
+        """
+        Get the standard deviation of a batch normalization layer.
+
+        Args:
+            bn_layer_name (str): Name of the batch normalization layer.
+
+        Returns:
+            Any: Standard deviation of the batch normalization layer.
+        """
+        return self.bn_params[bn_layer_name][2]
+
+    def get_num_bn_layers(self) -> int:
+        """
+        Get the number of batch normalization layers.
+
+        Returns:
+            int: Number of batch normalization layers.
+        """
+        return len(self.bn_params)
+
+    def get_bn_params(self,
+                      model: Any,
+                      bn_layer_types: Type[list]) -> dict:
+        """
+        Get the batch normalization parameters (mean and variance) for each batch normalization layer in the model.
+
+        Args:
+            model (Any): The model.
+            bn_layer_types (Type[list]): List of batch normalization layer types.
+
+        Returns:
+            dict: Dictionary mapping batch normalization layer names to their parameters.
+        """
+        raise NotImplemented
+
+
+
+class ActivationExtractor(object):
```

Comment:
Consider declaring it an abstract class? (correct me if I'm wrong, you do not expect an implementation of any of the methods here, but you do expect every class that inherit from it to implement all methods?)